src/Admin/Settings_Page.php

```diff
@@ -0,0 +1,81 @@
+<?php declare(strict_types=1);
+
+namespace Tribe\Tribe_Embed\Admin;
+
+class Settings_Page {
+
+	public const VIMEO_TOKEN = 'vimeo_access_token';
+
+	public function __construct() {
+		add_action( 'admin_menu', [ $this, 'add_menu' ] );
+		add_action( 'admin_init', [ $this, 'register_settings' ] );
+	}
+
+	/**
+	 * Register settings page under "Settings → Tribe Embeds"
+	 */
+	public function add_menu(): void {
+		add_options_page(
+			esc_html__( 'Tribe Embeds', 'tribe-embeds' ),
+			esc_html__( 'Tribe Embeds', 'tribe-embeds' ),
+			'manage_options',
+			'tribe-embeds-settings',
+			[ $this, 'render_page' ]
+		);
+	}
+
+	/**
+	 * Register Vimeo credentials.
+	 */
+	public function register_settings(): void {
+		register_setting(
+			'tribe_embeds_settings',
+			'tribe_embeds_vimeo_access_token',
+			[ 'sanitize_callback' => 'sanitize_text_field' ]
+		);
+	}
+
+	/**
+	 * Render the settings page.
+	 */
+	public function render_page(): void {
+		?>
+		<div class="wrap">
+			<h1><?php esc_html_e( 'Tribe Embeds – Vimeo Auth', 'tribe-embeds' ); ?></h1>
```

Comment:
Let's rename to `Tribe Embeds Settings` since this will contain more than just the Vimeo key setting.